src/utils.jl

```diff
@@ -0,0 +1,232 @@
+## Macros to help implement calculators
+
+
+"""
+    @generate_interface
+
+Generate complementary function for given function expression.
+This is mean to help defining the interface, so that you only need
+to define one of the interface methods for a given type of calculation
+(energy, forces, virial).
+
+# Example
+
+Generate `forces!` and `calculate(AtomsCalculators.Forces(), ...)` calls from `forces` definition
+
+```julia
+AtomsCalculators.@generate_interface function AtomsCalculators.forces(system, calculator::Main.MyType; kwargs...)
+    #definition here
+end
+```
+
+Generate `forces` and  `calculate(AtomsCalculators.Forces(), ...)` calls from `forces!` definition
+
+```julia
+AtomsCalculators.@generate_interface function AtomsCalculators.forces!(f::AbstractVector, system, calculator::Main.MyOtherType; kwargs...)
+    #definition here
+end
+```
+
+Generate `AtomsCalculators.potential_energy` call from `AtomsCalculators.calculate` call.
+
+```julia
+AtomsCalculators.@generate_interface function AtomsCalculators.calculate(::AtomsCalculators.Energy(), system, calculator::Main.MyType; kwargs...)
+    #definition here
+end
+```
+
+# Note
+
+When using this macro, you need to define your calculator type completely - `MyPkg.MyType` is fine
+but `MyType` is not!
+
+"""
+macro generate_interface(expr)
```

Comment:
I still feel along the lines that explicit is better than implicit, hence I would not make one super macro that tries to be smart, rather have different macros for the tasks of generating the calculator-based interface or the single-function-based interface.

Reply:
The is not much complications on having general macro versus specific ones. For user perspective it is easier to just use a single one and for implementation point a single one is in practice the same, so I would prefer a single general macro over several specific ones. But we can change this later, if there are some issues with it.

Reply:
ok for me.

Reply:
Ok, this is not a blocker for me to be fair, more my gut feeling. :smile: